design-documents/add-scope-to-cms-pages.md

```diff
@@ -0,0 +1,94 @@
+-   "child" pages have a different page ID than will be displayed in the URL, which could be confusing for 3rd party developers
+-   any externally developed modules that list pages would have to be updated to filter out "child" pages
+-   More complicated logic to load/save the desired page/store view content  
+
+##### Scenario: New Page
+When a new page is created it will default to the All Store Views scope, and the data will be saved to `cms_page` as it is now, with `parent_id` = null and `store_id` = 0.
+
+##### Scenario: Editing a new scope
+After a page is created, changing scope will attempt to load the content using `parent_id` = page_id and the selected  `store_id`. If none exists, the data for the  `page_id` and  `store_id=0` will be loaded. If changes are made and saved, a new  `cms_page` entry will be created to save data with a new  `page_id`, setting the  `parent_id` = parent  `page_id` and using the selected  `store_id`. Since `is_active` is a global attribute, its value will be copied from the parent page.
+
+Code will need to be created to always load the `is_active` value from the parent page, OR to always propagate any change in value to all child pages so it can be loaded directly from the child page.
+
+##### Scenario: Deleting a page
+Delete a page deletes all content associated with it for all store views.
+
+##### Scenario: Page Builder needs selected store view scope
+Page Builder components (and potentially other extensions) will need to be aware of the selected store view to render store-sensitive data (such as Product List). The store switcher component adds the chosen store_id to the application context and will be available as Page Builder currently expects (via StoreManager) without additional changes.
+
+#### Acceptance Criteria Fulfillment
+1. СMS Pages have Magento Scope.
+2.  Admin user with scope specific role is able to edit scope specific content
+3.  CMS Page is assigned to AllStoreviews by default.
+4.  Changing the drop-down value will cause the page to load the content for that specific storeview
+5.  CMS Page content attributes have storeview scope except:
+		- enable page - global
+		- store view - global
+6.  Changing the drop-down value to any value other than All Store Views will by default have the "Use Default Value" checkbox checked for all storeview specific attributes
+7.  ContentHeading is removed from the CMS Page Form(only new pages affected)
+8.  PageTitle content attribute value is copied to MetaTitle by default
+
+#### Component Dependencies
+- Store
+
+#### Extension Points and Scenarios
+Page Builder components (and potentially other extensions) will need to be aware of the selected store view to render store-sensitive data (such as Product List). The store switcher component adds the chosen store_id to the application context and will be available as Page Builder currently expects (via StoreManager) without additional changes.
+
+### Prototype or Proof of Concept
```

Comment:
This section can be removed if you are not going to provide prototype.

Reply:
if you're referring to the Use Default Value checkbox, this just causes the content to be copied from the Default Store View scope. Dynamic block currently functions this way.

Reply:
Done